pkg/operator/controller/extension/care/health.go

```diff
@@ -0,0 +1,200 @@
+		return nil, fmt.Errorf("failed to list controller installations: %w", err)
+	}
+
+	if exitCondition, err := h.healthChecker.CheckControllerInstallations(ctx, h.gardenClientSet.Client(), condition, controllerInstallations.Items, func(ci gardencorev1beta1.ControllerInstallation) bool {
+		return ci.Spec.RegistrationRef.Name == h.extension.Name
+	}, nil); err != nil {
+		return nil, fmt.Errorf("failed to check controller installations: %w", err)
+	} else if exitCondition != nil {
+		return exitCondition, nil
+	}
+
+	return ptr.To(v1beta1helper.UpdatedConditionWithClock(h.clock, condition, gardencorev1beta1.ConditionTrue, "ControllerInstallationsRunning", "All controller installations are healthy.")), nil
+}
+
+func (h *health) checkExtension(ctx context.Context, condition gardencorev1beta1.Condition) (*gardencorev1beta1.Condition, error) {
+	managedResource := &resourcesv1alpha1.ManagedResource{}
+	managedResourceName := helper.ExtensionRuntimeManagedResourceName(h.extension.Name)
+
+	if err := h.runtimeClient.Get(ctx, client.ObjectKey{Namespace: h.gardenNamespace, Name: managedResourceName}, managedResource); err != nil {
+		return nil, fmt.Errorf("failed to get managed resource %s: %w", managedResourceName, err)
+	}
+
+	if exitCondition := h.healthChecker.CheckManagedResource(condition, managedResource, nil); exitCondition != nil {
+		return exitCondition, nil
+	}
+
+	return ptr.To(v1beta1helper.UpdatedConditionWithClock(h.clock, condition, gardencorev1beta1.ConditionTrue, "ExtensionComponentsRunning", "All extension components are healthy.")), nil
+}
+
+func (h *health) checkExtensionAdmission(ctx context.Context, condition gardencorev1beta1.Condition) (*gardencorev1beta1.Condition, error) {
+	managedResourceNames := []string{
+		helper.ExtensionAdmissionVirtualManagedResourceName(h.extension.Name),
+		helper.ExtensionAdmissionRuntimeManagedResourceName(h.extension.Name),
+	}
+
+	for _, managedResourceName := range managedResourceNames {
+		managedResource := &resourcesv1alpha1.ManagedResource{}
+		if err := h.runtimeClient.Get(ctx, client.ObjectKey{Namespace: h.gardenNamespace, Name: managedResourceName}, managedResource); err != nil {
+			return nil, fmt.Errorf("failed to get managed resource %s: %w", managedResourceName, err)
+		}
+
+		if exitCondition := h.healthChecker.CheckManagedResource(condition, managedResource, nil); exitCondition != nil {
+			return exitCondition, nil
+		}
+	}
+
+	return ptr.To(v1beta1helper.UpdatedConditionWithClock(h.clock, condition, gardencorev1beta1.ConditionTrue, "ExtensionAdmissionComponentsRunning", "All extension admission components are healthy.")), nil
+}
+
+// ExtensionConditions contains all conditions of the extension status subresource.
+type ExtensionConditions struct {
+	controllerInstallationsHealthy *gardencorev1beta1.Condition
+	extensionHealthy               *gardencorev1beta1.Condition
+	extensionAdmissionHealthy      *gardencorev1beta1.Condition
```

Comment:
Is there any benefit of separating the conditions instead of having a single `Healthy` condition that combines all? In case of an error, it could still provide more details in the condition's message.

Reply:
The consequences and measurements are differently for each type of error.
- When controller-installations are not healthy, you might get issues on your seeds and they become unhealthy.
- When the extension is not healthy you'll get issues in your runtime cluster like with your backup bucket.
- When the extension admission is not healthy, some APIs in the Garden cluster might be broken.

For me this differentiation is helpful.